src/library/f3d_options.cxx

```diff
@@ -0,0 +1,71 @@
+#include "f3d_options.h"
+
+#include "F3DOperators.h"
+
+#include <map>
+
+namespace f3d
+{
+//----------------------------------------------------------------------------
+class options::F3DInternals
+{
+public:
+  std::map<std::string, std::string> Options;
+};
+
+//----------------------------------------------------------------------------
+options::options() : Internals(new options::F3DInternals) {};
+options::~options() = default;
+
+//----------------------------------------------------------------------------
+template <typename T>
+void options::set(const std::string &name, const T& value)
+{
+  std::stringstream ss;
+  ss << value;
+  this->Internals->Options[name] = ss.str();
+}
+
+//----------------------------------------------------------------------------
+template void options::set<bool>(const std::string &name, const bool& value);
```

Comment:
I don't think it's necessary to write `<bool>`, it should be deduced by the second argument.